core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java

```diff
@@ -250,7 +250,12 @@ public static void updateTableInfo(TableInfo tableInfo) {
   public static CarbonTable buildTable(
       String tablePath,
       String tableName,
-      Configuration configuration) throws IOException {
+      Configuration configuration,
+      boolean isFile) throws IOException {
+    if (isFile) {
+      int index = tablePath.lastIndexOf("/");
```

Comment:
What is the File is in sub folder ?  taking table path as subfolder path is ok ?

Reply:
It's ok. subfolder path is ok.